### DISCUSSION TITLE
mapping keys in nvim-cmp

### DISCUSSION THREAD

**Discussion by tudorjnu** [LazyVim/LazyVim#1052] [Q&A] [answered]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.0-dev

### Operating system/version

Pop!_OS 22.04 LTS

### Describe the bug

I tried setting the confirmation key map for cmp using the following code, which is in `.config/nvim/lua/plugins/<custom_settings>.lua`:

```lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = cmp.mapping.preset.insert(vim.tbl_deep_extend("force", opts.mapping, {
        ["C-j"] = cmp.mapping.confirm({ select = true }),
      }))
    end,
  }
}
```

However, besides the key map not being set, it seems like it sets a different key map altogether (`cmp.utils.keymap.set_map`). 

### Steps To Reproduce

Add the piece of code within a file within `lua/plugins/<filename>.lua

### Expected Behavior

The keymap to be bound to cmp confirm.

**Comment by radvil** [comment]
IDK, if this will work, but try to return the opts. It might look like this.
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = cmp.mapping.preset.insert(vim.tbl_deep_extend("force", opts.mapping, {
        ["C-j"] = cmp.mapping.confirm({ select = true }),
      }))
      return opts
    end,
  }
}
```

**Comment by tudorjnu** [comment]
Thank you, @radvil! Unfortunately, it doesn't work.

**Comment by radvil** [comment]
what is that "c-j" btw ?

**Comment by Gandharikos** [comment, marked as answer]
@tudorjnu 
```
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping["<C-j>"] = cmp.mapping.confirm({ select = true })

    end,
  }
}
```

**Comment by folke** [comment]
converted to a discussion since this is a user question.

**Comment by tudorjnu** [comment]
Sorry for that! I missed that `"<C-j>"`. Thank you!
